public/marked_entity_files.php

```diff
@@ -113,18 +154,34 @@
                 </td>
 
                 <!-- TODO: Should we apply some sort of transformation to the file ID -->
-                <td><a href='<?php echo "download.php?file_id={$row->attachment->file_id}" ?>'>Download</a></td>
                 <td>
+                    <?php
+                        $may_read = $row->get_permission_for_user($_SESSION["current_user"]->id, "read");
+                    ?>
+                    <button onclick="location.href='<?php echo "download.php?file_id={$row->attachment->file_id}" ?>'" type="button" <?php if (!$may_read) {
+                        echo "disabled";
+                    } ?>>Download</button>
```

Comment:
This can be abused since people can access the link even if the button is disabled 😂 are we willing to overlook this?

Reply:
we can add a permission check in download.php if we really wanted to

Reply:
cant you send the permissions through another GET variable?
youre already sending file_id

Reply:
so then if they navigate to the link without going through the proper routes, the GET variable for permissions would be null or something

Reply:
WELLLLLLL then the malicious user can just pass in any permission that he wants 😂